client/zeth/core/input_hasher.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2015-2021 Clearmatics Technologies Ltd
+#
+# SPDX-License-Identifier: LGPL-3.0+
+
+from zeth.core.mimc import MiMCBase
+from typing import List
+
+
+# Default seed, generated as:
+#   zeth.core.mimc._keccak_256(
+#       zeth.core.mimc._str_to_bytes("clearmatics_hash_seed"))
+DEFAULT_IV_UINT256 = \
+    13196537064117388418196223856311987714388543839552400408340921397545324034315
+
+
+class InputHasher:
+    """
+    Hash a series of field elements via the Merkle-Damgard construction on a
+    MiMC compression function. Note that since this function only accepts whole
```

Comment:
> on a MiMC compression function

Do I understand correctly that by "a MiMC compression function" you built such a function from the MiMC block cipher via Miyaguchi-Preneel (like in the Zeth specs)?
Let's make sure that such instantiations are formally documented before inclusion in the code. Happy to move forward with this PR when it is ready to go, but I would add an "EXPERIMENTAL" mention here until we get this one on paper.

Reply:
added a mention that this is experimental.